### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by prateekma**
- [x] Update README with information on running integration tests
- [x] Make integration tests launch Simulation GUI on non-CI platforms
- [x] Figure out why Windows is not working and fix it

Note to future self when merging: Add co-authorship to @Piphi5

### CODE REVIEW

File: src/integrationtest/native/cpp/SimulationTest.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+#include "networktables/NetworkTableValue.h"
+#include "sysid/analysis/AnalysisManager.h"
+#include "sysid/analysis/AnalysisType.h"
+#include "sysid/telemetry/TelemetryManager.h"
+
+#ifdef _WIN32
+#define SEPARATOR "\\"
+#define LAUNCHSIM "gradlew simulateCpp"
+#else
+#define SEPARATOR "/"
```
Comment (Piphi5) [inline src/integrationtest/native/cpp/SimulationTest.cpp:27]: Move `SEPARATOR` to `TelemetryManager.h`.

File: src/main/native/cpp/telemetry/TelemetryManager.cpp [hunk 2]
```diff
@@ -130,14 +135,16 @@ void TelemetryManager::SaveJSON(wpi::StringRef location) {
   ss << std::put_time(&tm, "%Y%m%d-%H%M");
```
Comment (Piphi5) [inline src/main/native/cpp/telemetry/TelemetryManager.cpp:135]: Add the `SEPARATOR` into the ss:
```
ss << SEPARATOR;
ss << "sysid";
```

File: sysid_data20210213-1223.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+"test": "Drivetrain",
+"units": "Meters",
+"unitsPerRotation": 1.0
+}
```
Comment (Piphi5) [inline sysid_data20210213-1223.json:5]: Is this supposed to be here?

Comment (prateekma) [reply sysid_data20210213-1223.json:5]: Nope, I'll remove it in the next commit.

**Pull request merged**